contract/src/views.rs

```diff
@@ -0,0 +1,141 @@
+use crate::*;
+
+#[near_bindgen]
+impl LinkDropProxy {
+    /// Returns the balance associated with given key. This is used by the NEAR wallet to display the amount of the linkdrop
+    pub fn get_key_balance(&self, key: PublicKey) -> U128 {
+        let account_data = self.data_for_pk
+            .get(&key)
+            .expect("Key missing");
+        (account_data.balance.0).into()
+    }
+
+    /*
+        CUSTOM
+    */
+    
+    /// Query for the total supply of linkdrops on the contract
+    pub fn key_total_supply(
+        &self
+    ) -> U128 {
+        //return the length of the data_for_pk set
+        U128(self.data_for_pk.len() as u128)
+    }
+
+    /// Paginate through all active keys on the contract and return a vector of key info.
+    pub fn get_keys(
+        &self, 
+        from_index: Option<U128>, 
+        limit: Option<u64>
+    ) -> Vec<KeyInfo> {
+        //where to start pagination - if we have a from_index, we'll use that - otherwise start from 0 index
+        let start = u128::from(from_index.unwrap_or(U128(0)));
+
+        //iterate through each key using an iterator
+        self.data_for_pk.keys()
+            //skip to the index we specified in the start variable
+            .skip(start as usize) 
+            //take the first "limit" elements in the vector. If we didn't specify a limit, use 50
+            .take(limit.unwrap_or(50) as usize) 
+            //we'll map the public keys which are strings into KeyInfos
+            .map(|pk| self.get_key_information(pk.clone()))
+            //since we turned the keys into an iterator, we need to turn it back into a vector to return
+            .collect()
+    }
+
+    
+    /// Returns the total supply of active keys for a given funder
+    pub fn key_supply_for_funder(
+        &self, 
+        account_id: AccountId,
+    ) -> U128 {
+        //get the set of keys for the passed in funder
+        let keys_for_owner = self.keys_for_funder.get(&account_id);
+
+        //if there is some set of keys, we'll return the length as a U128
+        if let Some(keys_for_owner) = keys_for_owner {
+            U128(keys_for_owner.len() as u128)
```

Comment:
curious why converting to U128, is it just for the string serialization?

Reply:
Yea it was for string serialization. This is the way I've always done it but if it's non optimal, lemme know and I can change it

Reply:
could have just used `U64` to not convert to 128 bit, but it's very trivially better so doesn't matter